tests/cstor/volume/provision_bulk_test.go

```diff
@@ -0,0 +1,263 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package volume
+
+import (
+	"strconv"
+	"strings"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	apis "github.com/openebs/maya/pkg/apis/openebs.io/v1alpha1"
+	pvc "github.com/openebs/maya/pkg/kubernetes/persistentvolumeclaim/v1alpha1"
+	sc "github.com/openebs/maya/pkg/kubernetes/storageclass/v1alpha1"
+	spc "github.com/openebs/maya/pkg/storagepoolclaim/v1alpha1"
+	"github.com/openebs/maya/tests/cstor"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var _ = Describe("[cstor] TEST BULK VOLUME PROVISIONING", func() {
+	var (
+		err           error
+		pvcNamePrefix = "cstor-volume-claim-"
+		bulkCount     = 10
+		pvcObjList    *v1.PersistentVolumeClaimList
+		pvcObjTpl     *v1.PersistentVolumeClaim
+	)
+
+	When("creating resources required by test", func() {
+		It("should create a spc and storageclass", func() {
+
+			By("building spc")
+			spcObj = spc.NewBuilder().
+				WithGenerateName(spcName).
+				WithDiskType(string(apis.TypeSparseCPV)).
+				WithMaxPool(cstor.PoolCount).
+				WithOverProvisioning(false).
+				WithPoolType(string(apis.PoolTypeStripedCPV)).
+				Build().Object
+
+			By("creating above spc")
+			spcObj, err = ops.SPCClient.Create(spcObj)
+			Expect(err).To(BeNil(), "while creating spc {%s}", spcName)
+
+			By("verifying healthy csp count")
+			cspCount := ops.GetHealthyCSPCountEventually(spcObj.Name, cstor.PoolCount)
+			Expect(cspCount).To(Equal(true), "while checking csp health status")
+
+			By("building cas config with above spc")
+			CASConfig := strings.Replace(
+				openebsCASConfigValue, "$spcName", spcObj.Name, 1,
+			)
+			CASConfig = strings.Replace(
+				CASConfig, "$count", strconv.Itoa(cstor.ReplicaCount), 1,
+			)
+			annotations[string(apis.CASTypeKey)] = string(apis.CstorVolume)
+			annotations[string(apis.CASConfigKey)] = CASConfig
+
+			By("building a storageclass")
+			scObj, err = sc.NewBuilder().
+				WithGenerateName(scName).
+				WithAnnotations(annotations).
+				WithProvisioner(openebsProvisioner).Build()
+			Expect(err).ShouldNot(
+				HaveOccurred(),
+				"while building storageclass {%s}", scName,
+			)
+
+			By("creating above storageclass")
+			scObj, err = ops.SCClient.Create(scObj)
+			Expect(err).To(BeNil(), "while creating storageclass {%s}", scName)
+		})
+	})
+
+	// Actual tests begin here !!!
+	When("cstor pvcs are created", func() {
+		It("should create cstor volumes", func() {
+
+			By("building a pvc template")
+			pvcObjTpl, err = pvc.NewBuilder().
+				WithGenerateName(pvcNamePrefix).
+				WithNamespace(nsObj.Name).
+				WithStorageClass(scObj.Name).
+				WithAccessModes(accessModes).
+				WithCapacity(capacity).
+				WithLabels(
```

Comment:
can we make use of this label to filter later

Reply:
No as the this label is for pvcs and we are fetching the pods. Pods don't have this label.